src/commands/check.rs

```diff
@@ -119,9 +124,61 @@ pub async fn check(args: CheckArgs) -> anyhow::Result<()> {
         );
     }
 
-    let remote_file = Url::parse(&file).is_ok();
+    let remote_file = Url::parse(file).is_ok();
+
+    if !exceptions.is_empty() {
+        let analysis_rules: HashSet<String> = analysis::rules()
+            .iter()
+            .map(|r| r.id().to_lowercase())
+            .collect();
+        let lint_rules: HashSet<String> = lint::rules()
+            .iter()
+            .map(|r| r.id().to_lowercase())
+            .collect();
+        let all_rules: HashSet<_> = analysis_rules.union(&lint_rules).collect();
+
+        let unknown_exceptions: Vec<&str> = exceptions
+            .iter()
+            .filter(|rule| !all_rules.contains(&rule.to_lowercase()))
+            .map(|rule| rule.as_str())
+            .collect();
+
+        if !unknown_exceptions.is_empty() {
+            let (config, writer) =
+                get_display_config(args.common.report_mode, args.common.no_color);
+            let mut w_lock = writer.lock();
+            let files: SimpleFiles<String, String> = SimpleFiles::new();
+
+            let message = if unknown_exceptions.len() == 1 {
+                format!(
+                    "ignoring unknown rule provided via --except: '{}'",
+                    unknown_exceptions[0]
+                )
+            } else {
+                format!(
+                    "ignoring unknown rules provided via --except: '{}'",
+                    unknown_exceptions
+                        .iter()
+                        .map(|r| format!("`{r}`"))
+                        .collect::<Vec<_>>()
+                        .join(", ")
+                )
+            };
+
+            let warning = codespan_reporting::diagnostic::Diagnostic::warning()
+                .with_message(message)
+                .with_code("UnknownExceptionRule")
```

Comment:
I think this should be removed. A user might see this and expect to be able to `--except UnknownExceptionRule` which won't work.

So no `with_code()` here is my thinking.